commands/volume-create/volume-create.go

```diff
@@ -20,46 +20,71 @@ import (
 type Command struct {
 }
 
-func (c *Command) volumeCreate(w http.ResponseWriter, r *http.Request) {
-
-	var msg volume.VolCreateRequest
-
-	e := utils.GetJSONFromRequest(r, &msg)
+func validateVolumeCreateRequest(msg *volume.VolCreateRequest, r *http.Request) *client.GenericJSONResponse {
+	e := utils.GetJSONFromRequest(r, msg)
 	if e != nil {
-		client.SendResponse(w, -1, 422, errors.ErrJSONParsingFailed.Error(), 422, "")
-		return
+		log.Error("Invalid JSON Request")
+		return client.FormResponse(-1, 422, errors.ErrJSONParsingFailed.Error(), "")
 	}
 
 	if msg.Name == "" {
 		log.Error("Volume name is empty")
-		client.SendResponse(w, -1, http.StatusBadRequest, errors.ErrEmptyVolName.Error(), http.StatusBadRequest, "")
-		return
+		return client.FormResponse(-1, http.StatusBadRequest, errors.ErrEmptyVolName.Error(), "")
 	}
 	if len(msg.Bricks) <= 0 {
 		log.Error("Brick list is empty")
-		client.SendResponse(w, -1, http.StatusBadRequest, errors.ErrEmptyBrickList.Error(), http.StatusBadRequest, "")
-		return
+		return client.FormResponse(-1, http.StatusBadRequest, errors.ErrEmptyBrickList.Error(), "")
 	}
+	return nil
+
+}
+
+func createVolume(msg *volume.VolCreateRequest) *volume.Volinfo {
+	vol := volume.NewVolumeEntry(msg)
+	return vol
+}
+
+func (c *Command) volumeCreate(w http.ResponseWriter, r *http.Request) {
+
+	msg := new(volume.VolCreateRequest)
 
+	rsp := validateVolumeCreateRequest(msg, r)
+	if rsp != nil {
+		client.SendResponse(w, http.StatusBadRequest, rsp)
+		return
+	}
 	if context.Store.VolumeExists(msg.Name) {
```

Comment:
You've changed all other context.Store.Vol* functions to volume.* functions but missed this.